app/controllers/api/v1/posts_controller.rb

```diff
@@ -3,13 +3,39 @@
 module Api
   module V1
     class PostsController < Api::V1::BaseController
+      before_action :set_user, only: %i[index create]
+      before_action :verify_current_user, only: :create
+
       def index
-        @posts = User.find(params[:user_id]).posts
+        @posts = @user.posts
       end
 
       def show
         @post = Post.find(params[:id])
       end
+
+      def create
+        @post = @user.posts.new(post_params)
+        return unless @post.save
```

Comment:
Here you could probably add the logic that checks if it's necessary to respond with bad_request